blockchain-albatross/src/history_store/history_store.rs

```diff
@@ -597,32 +629,33 @@ impl HistoryStore {
             }
         };
 
-        let mut leaf_indices = vec![];
+        let mut leaf_hashes = vec![];
 
-        // Seek to the first leaf index at the given transaction hash.
-        let mut cursor = txn.cursor(&self.leaf_idx_db);
-        let leaf_index = match cursor.seek_key::<Blake2bHash, u32>(tx_hash) {
-            None => return leaf_indices,
+        // Seek to the first leaf hash at the given block number.
+        let mut cursor = txn.cursor(&self.block_db);
+        let leaf_hash = match cursor.seek_key::<u32, Blake2bHash>(&block_number) {
+            None => return leaf_hashes,
             Some(v) => v,
         };
 
-        leaf_indices.push(leaf_index);
+        leaf_hashes.push(leaf_hash);
 
         loop {
-            // Get next leaf index.
-            match cursor.next_duplicate::<Blake2bHash, u32>() {
-                Some((_, leaf_index)) => leaf_indices.push(leaf_index),
+            // Get next leaf hash.
+            match cursor.next_duplicate::<u32, Blake2bHash>() {
+                Some((_, leaf_hash)) => leaf_hashes.push(leaf_hash),
                 None => break,
             };
         }
 
-        leaf_indices
+        leaf_hashes
     }
 
-    /// Returns a vector containing all leaf hashes corresponding to the given block number.
-    fn get_leaf_hash_by_block(
+    /// Returns a vector containing all transaction (no inherents) hashes corresponding to the given
+    /// address.
+    pub fn get_tx_hashes_by_address(
         &self,
-        block_number: u32,
+        address: &Address,
         txn_option: Option<&Transaction>,
     ) -> Vec<Blake2bHash> {
```

Comment:
The vec returned by this method can get veeeeery long for addresses with loads of transactions, as for example pools will, or games like Sunset Cyberspace.
I think it makes sense that this method gets a parameter `limit` (or `max` or similar), so that it stops seeking after that amount of hashes has been found.
Will hashes be returned in the reverse order they have been inserted, so in reverse-historical order?

Reply:
I don't know in what order they get fetched from the database, that's a good question. Let me try to find out.